server/src/main/java/org/elasticsearch/transport/RemoteClusterAware.java

```diff
@@ -73,6 +79,10 @@ protected RemoteClusterAware(Settings settings) {
         for (String index : requestIndices) {
             int i = index.indexOf(RemoteClusterService.REMOTE_CLUSTER_INDEX_SEPARATOR);
             if (i >= 0) {
+                if (isRemoteClusterClientEnabled == false) {
+                    assert remoteClusterNames.isEmpty() : remoteClusterNames;
+                    throw new IllegalStateException("node [" + nodeName + "] does not have the remote cluster client role enabled");
```

Comment:
I am not too sure about `IllegalStateException` here. It's more like a user error as they should rather send the request to another node? Maybe `IllegalArgumentException` would be more appropriate but I am not 100% happy with it either

Reply:
Yes, I think we should consider it a bad request. I've adjusted it in https://github.com/elastic/elasticsearch/pull/60351/commits/78ba655192c7788b0082da15526f0ba0bcc79691.